### DISCUSSION TITLE
OMR prefix

### DISCUSSION THREAD

**Discussion by seancmalloy** [openthread/openthread#9332] [Q&A] [answered]
does anyone know why im receiving this error when i try to unsuccessfully add matter over thread device with home assistant?

otbr-agent[179]: 00:09:06.432 [I] RoutingManager: - RIO fd12:f994:6b9f::/48 - ignore since not a valid OMR prefix

**Comment by jwhui** [comment, marked as answer]
This occurs when processing Route Information Options contained within Route Advertisement messages received on the Adjacent Infrastructure Link (e.g. Wi-Fi / Ethernet).

https://github.com/openthread/openthread/blob/a361aebc8b566ae9225c8163025550f2867bc112/src/core/border_router/routing_manager.cpp#L983

Off Mesh Routable (OMR) prefixes are required to have 64 bit length. Any other prefix length is ignored for the purpose of Thread routing.

https://github.com/openthread/openthread/blob/a361aebc8b566ae9225c8163025550f2867bc112/src/core/border_router/routing_manager.cpp#L840-L844

**Comment by seancmalloy** [comment]
thanks for the input.

**Reply by jwhui** [reply]
The log by itself is not an error - simply receiving RIOs for other IPv6 prefixes may be expected depending on your network setup.

But to dig deeper, you can output the information being advertised by other routers on the link:
```
sudo rdisc6 <netif>
```

**Comment by seancmalloy** [comment]
thanks. ill look into it. still trying to get familiar with matter over thread.
